phase0/CRC.md

```diff
@@ -0,0 +1,205 @@
+* depleteHealth: Method, updates health after being attacked 
+
+### Collaborators
+Player, Enemy, Defender
+
+---
+
+# `class Player`
+
+### Responsibilities
+* Spawnable_defenders: Attribute, dictionary of level of defenders and their corresponding integer of number of defenders the player can spawn.  
+* Build_wall(position): Method to use inventory to build wall at a certain position on the map
+* Spawn_defender(position): Method to spawn defender at a certain position on the map
+
+### Collaborators
+Player, Enemy, Defender, Weaponable, Collidable, IndividualCharacter
+
+---
+
+# `class Enemy`
+
+### Responsibilities
+* Path: Attribute, ArrayList, the path the enemy is meant to follow
+* Attack: Method, attacks defense on the map
+
+### Collaborators
+IndividualCharacter, Player, Defender, Map, Weaponable, Collidable
+
+---
+
+# `class Defender`
+
+### Responsibilities
+* Attack: Method, attacks enemies on the map
+
+### Collaborators
+IndividualCharacter, Player, Enemy, Map, Weaponable, Collidable
+
+---
+
+# `class Tower (?)`
+
+### Responsibilities
+* Position: Attribute, array with x, y, where on the map the Tower is currently located
+* Health: Attribute, total health of the tower
+* updateHealth: method, updates the health of the tower to reflect damage taken
+
+### Collaborators
+Enemy, Map, Collidable
+
+---
+
+# `class Map`
+
+### Responsibilities
+* All_positions: Attribute, list of all objects and their positions
+
+
+### Collaborators
+Player, Enemy, Defender, Tower
+
+---
+
+# `interface Weaponable`
+
+### Responsibilities
+* Hit(attacker, victim, weapon): Method that depletes victim health by weapon’s attackPower
+
+
+### Collaborators
+Player, Enemy, Defender
+
+---
+
+# `interface Collidable`
+
+### Responsibilities
+* Collide: Method that collides two objects whenever in the same position and depletes health according to objects clashed. 
+
+
+### Collaborators
+Player, Enemy, Defender, Tower
+
+---
+
+# `abstract class Weapon`
+
+### Responsibilities
+* Grade: Attribute, essentially the level of the weapon
+* attackPower: The amount of damage the weapon does
+
+### Collaborators
+Player, Enemy, Defender, Collidable
+
+---
+
+<!-- _class: lead --># Use Case Classes
+
+---
+
+# `class CharacterManager`
+
+### Responsibilities
+* moveCharacter: method, takes directional commands from keyboard input and moves character
+* attackObject: method, attacks nearest object when the range <= distance to that object (for NPCs)
+* takeDamage/depleteHealth: method, takes health away from the character when hit
+
+### Collaborators
+Player, Enemy, Defender, Collidable 
+
+---
+
+# `class gameManager (?)`
+
+### Responsibilities
+* Spawn: Method, spawns enemies at position on the map
+* deleteCharacter: removes a character from the map when health = 0
+
+
+### Collaborators
+Player, Enemy, Defender, Map
+
+---
+
+<!-- _class: lead -->
+# Controller Classes
+
+---
+
+# `class InputHandler`
```

Comment:
rename to `KeyboardInputHandler` and mutate an entity class like `ControlState` to allow getting controls from other places, like the command line for example